### PR TITLE
add quantization recipies fro IEMCAP, CV, LibriSpeech and LJSpeech

### PR DESCRIPTION
**Pull request by poonehmousavi**
## What does this PR do?

add/update quantization recipes for IEMOCAP, CommonVoive, LJSpeech, LibriSpeech

Fixes #<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

File: speechbrain/utils/kmeans.py
Comment (Chaanks) [inline speechbrain/utils/kmeans.py:1]: I have one comment regarding the function `accumulate_and_extract_features`. I think the `features_list` parameter can be removed and features returned directly at the end of the function.
Following my previous comment, `features_list = []` can be removed in favor to `features_list = accumulate_and_extract_features(...)`.

File: speechbrain/utils/kmeans.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            # extract features from the SSL model
+            accumulate_and_extract_features(
+                batch, features_list, ssl_model, ssl_layer_num, device
+            )
+
+
+def save_model(model, checkpoint_path):
+    """Save a  Kmeans model .
+    Args:
+        model (MiniBatchKMeans): the  kmenas model to be saved.
```
Comment (Chaanks) [inline speechbrain/utils/kmeans.py:128]: typo in docstring

File: speechbrain/utils/kmeans.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    model,
+    train_set,
+    ssl_model,
+    ssl_layer_num,
+    kmeans_batch_size=1000,
+    device="cpu",
+):
+    """Train a  Kmeans model .
+    Args:
+        model (MiniBatchKMeans): the initial kmenas model for training.
```
Comment (Chaanks) [inline speechbrain/utils/kmeans.py:104]: typo in docstring

**Comment by Chaanks** [comment]
This isn't directly related to the current PR, but I've noticed that all discrete interfaces (like HuBERT, Wav2Vec 2.0, and WavLM) only support loading pretrained K-means models from HuggingFace. It might be useful to also support loading models locally. What are your thoughts on this?
@poonehmousavi  @mravanelli

**Comment by poonehmousavi** [comment]
> This isn't directly related to the current PR, but I've noticed that all discrete interfaces (like HuBERT, Wav2Vec 2.0, and WavLM) only support loading pretrained K-means models from HuggingFace. It might be useful to also support loading models locally. What are your thoughts on this? @poonehmousavi @mravanelli

there is cache_dir=cache_dir param when loading the model.. if the model exists there, it tries to fetch the model from that dir instead of HF

File: speechbrain/utils/kmeans.py
Comment (poonehmousavi) [reply speechbrain/utils/kmeans.py:1]: Not sure. if I understand the idea you suggested..I need to accumulate the feat from different batch somewhere....

Comment (Chaanks) [reply speechbrain/utils/kmeans.py:1]: Sorry I missed the fact that you extend the list inside the `accumulate_and_extract_features` function.

**Comment by mravanelli** [comment]
Thank you @poonehmousavi  and @Chaanks for your efforts on this PR. I did a code inspection and ran tests,
Here are my comments:

1. **Recipe Tests:**
   - Recipe tests are currently failing. You can test this using the following command:
     ```bash
     python -c 'from tests.utils.recipe_tests import run_recipe_tests; print("TEST FAILED!") if not(run_recipe_tests(filters_fields=["Task"], filters=[["quantization"]], do_checks=True, run_opts="--device=cuda")) else print("TEST PASSED")'
     ```
     I'll run a full recipe once the recipe tests are fixed.

2. **Vocoders Code:**
   - It appears that the code for the Vocoders by @Chaanks  is needed. Could we create a separate PR for this?

3. **Minor Issues:**

   - `speechbrain/utils/kmeans.py`:
     - Ensure that the docstrings are formatted consistently with others in the SpeechBrain codebase.
     - Consider adding runnable examples in some docstrings, such as for the train function.

   - `LibriSpeech/quantization`:
     - A README file is missing. Please add it to enhance documentation.

   - `recipes/CommonVoice/quantization/README.md`, `recipes/IEMOCAP/quantization/README.md`, `recipes/LJSpeech/quantization/README.md`:
     - Fix typo: `gneration` => `generation`.
     - The dropbox links point to an empty folder.

**Comment by poonehmousavi** [comment]
1. Vocoders Code: @Chaanks  is going to create a separate PR.
2. LibriSpeech/quantization already has a readme.md file
3. speechbrain/utils/kmeans.py : Formatiing is fixed. Adding example for train function need to pass the dataloader object. I am not sure what is the easy way to generate some random dataset and then pass it to dataloader without the need to create custom Dataset Class.
4.  All other suggestions are applied.

**Comment by mravanelli** [comment]
I did a final code inspection and ran the recipe tests. All looks now good to me. Thank you @poonehmousavi and @Chaanks for working on it!
